packages/grafana-data/src/utils/valueMappings.ts

```diff
@@ -1,104 +1,88 @@
-import { ValueMapping, MappingType, ValueMap, RangeMap } from '../types';
-
-type TimeSeriesValue = string | number | null;
-
-const addValueToTextMappingText = (
-  allValueMappings: ValueMapping[],
-  valueToTextMapping: ValueMap,
-  value: TimeSeriesValue
-) => {
-  if (valueToTextMapping.value === undefined) {
-    return allValueMappings;
-  }
-
-  if (value === null && isNullValueMap(valueToTextMapping)) {
-    return allValueMappings.concat(valueToTextMapping);
-  }
-
-  let valueAsNumber, valueToTextMappingAsNumber;
-
-  if (isNumeric(value as string) && isNumeric(valueToTextMapping.value)) {
-    valueAsNumber = parseFloat(value as string);
-    valueToTextMappingAsNumber = parseFloat(valueToTextMapping.value as string);
-
-    if (valueAsNumber === valueToTextMappingAsNumber) {
-      return allValueMappings.concat(valueToTextMapping);
+import { ValueMapping, MappingType, ValueMappingResult, SpecialValueMatch } from '../types';
+
+export function getValueMappingResult(valueMappings: ValueMapping[], value: any): ValueMappingResult | null {
```

Comment:
Would a FieldRegistry for ValueMappings be a nicer approach?

Reply:
maybe, but do no foresee many of these mapping types, think these 3 will suffice (famous last words).